CONTRIBUTING.md

```diff
@@ -38,3 +38,131 @@ are conflicts (even trivially resolvable ones).
 
 PGP signing commits is strongly encouraged. That should explain why
 the previous paragraph is here.
+
+# [Code of Conduct (22/C4.1)](http://rfc.zeromq.org/spec:22)
+
+## License
+
+Copyright (c) 2009-2015 Pieter Hintjens.
+
+This Specification is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 3 of the License, or (at your option) any later version.
+
+This Specification is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program; if not, see <http://www.gnu.org/licenses>.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+## Goals
+
+C4 is meant to provide a reusable optimal collaboration model for open source software projects. It has these specific goals:
+
+- To maximize the scale and diversity of the community around a project, by reducing the friction for new Contributors and creating a scaled participation model with strong positive feedbacks;
+- To relieve dependencies on key individuals by separating different skill sets so that there is a larger pool of competence in any required domain;
+- To allow the project to develop faster and more accurately, by increasing the diversity of the decision making process;
+- To support the natural life cycle of project versions from experimental through to stable, by allowing safe experimentation, rapid failure, and isolation of stable code;
+- To reduce the internal complexity of project repositories, thus making it easier for Contributors to participate and reducing the scope for error;
+- To enforce collective ownership of the project, which increases economic incentive to Contributors and reduces the risk of hijack by hostile entities.
+
+## Design
+
+### Preliminaries
+
+- The project SHALL use the git distributed revision control system.
+- The project SHALL be hosted on github.com or equivalent, herein called the "Platform".
+- The project SHALL use the Platform issue tracker.
```

Comment:
>The project SHALL use the git distributed revision control system.

git; check.

>The project SHALL be hosted on github.com or equivalent, herein called the "Platform".
>or equivalent
>>This enforces the use of the same software for both the source repo and bug tracker
>>same software for both the source repo and bug tracker
>>In any case, this is not responsive to the comment

Yes, this was responsive to the comment. Same software? No. Trac or "equivalent" does not imply git. Same hosting platform for both source and issue tracker? That depends on the definition of both ["equivalent"](https://git.wiki.kernel.org/index.php/GitHosting) and "hosted". Hosting a vanilla git repo and Trac on the same machine would be the "Platform" and we would still use the Platform issue tracker (in this case, Trac).

A simple amendment to the last line could clarify any confusion.